fluent-validator/src/test/java/com/baidu/unbiz/fluentvalidator/FluentValidatorWIthoutAnnotationsTest.java

```diff
@@ -0,0 +1,96 @@
+package com.baidu.unbiz.fluentvalidator;
+
+import com.baidu.unbiz.fluentvalidator.annotation.FluentValid;
+import com.baidu.unbiz.fluentvalidator.util.Function;
+import com.google.common.collect.Lists;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class FluentValidatorWIthoutAnnotationsTest {
+
+    class Passenger {
+        String name;
+        Boolean hasSeatbeltOn;
+
+        public Passenger(String name, Boolean hasSeatbeltOn) {
+            this.name = name;
+            this.hasSeatbeltOn = hasSeatbeltOn;
+        }
+    }
+
+    class Car {
+        private String make;
+        private List<Passenger> passengers = Lists.newArrayList();
+
+        public void addPassenger(Passenger passenger) {
+            passengers.add(passenger);
+        }
+    }
+
+    class SeatBeltValidator extends ValidatorHandler<Boolean> implements Validator<Boolean> {
+        private String name;
+
+        public SeatBeltValidator(String name) {
+            this.name = name;
+        }
+
+        public boolean validate(ValidatorContext context, Boolean hasSeatbeltOn) {
+            if (!hasSeatbeltOn) {
+                ValidationError error = new ValidationError();
+                error.setErrorCode(123);
+                error.setErrorMsg("passenger " + name + " must have their seat belt on");
+                context.addError(error);
+            }
+
+            return hasSeatbeltOn;
+        }
+    }
+
+    class SeatBeltValidatorWithoutError extends ValidatorHandler<Boolean> implements Validator<Boolean> {
+        public boolean validate(ValidatorContext context, Boolean hasSeatbeltOn) {
+            return hasSeatbeltOn;
+        }
+    }
+
+    class PassengerValidator extends ValidatorHandler<Passenger> implements Validator<Passenger> {
+        public boolean validate(ValidatorContext context, final Passenger passenger) {
+            return context.validateWith(new Function<FluentValidator, FluentValidator>() {
```

Comment:
I do not think that Closure or ValidationChain help in this case. We are trying to achieve validation of an object *and* its children without having to specify the relationships between the children at the top-level FluentValidator. We're also seeking to programmatically describe the validation rules (without annotations).